### PR TITLE
[Doc] typo fix build command line in README.md

### PR DESCRIPTION
**Pull request by zbtzbtzbt**
Changelog category (leave one):
- Documentation (changelog entry is not required)

Changes:
typo fix build command line in README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ git clone --recursive https://github.com/ByConity/ByConity.git byconity
 ### 3. Build
 
 ```
-cd byconity
+cd ByConity
```
Comment (hustnn) [inline README.md:63]: In step 2, it renamed to byconity like this 
`git clone --recursive https://github.com/ByConity/ByConity.git byconity`

If changing in this way, maybe we can remove byconity above.

Comment (zbtzbtzbt) [reply README.md:63]: Yours is also a good way.
now many open source projects are case-sensitive when cloning like CH,
```
git clone --recursive https://github.com/ClickHouse/ClickHouse.git
cd ClickHouse
```
I think we can follow this way~

Comment (hustnn) [reply README.md:63]: sure, can you help to change to that way? Thanks.

Comment (zbtzbtzbt) [reply README.md:63]: done.

**Review by hustnn — APPROVED** [review]

**Pull request merged**